pkg/minio/api-handlers.go

```diff
@@ -527,6 +530,8 @@ func (h objectAPIHandlersWrapper) ListBucketsHandler(w http.ResponseWriter, r *h
 	if err := h.bucketPrefixSubstitutionWithoutObject(w, r, "ListBuckets"); err != nil {
 		return
 	}
-
+	if !h.checkBucketExistence(r) {
```

Comment:
I'm not sure if this is needed.
In our case we should return a list of top level prefixes from the real bucket.
ListBuckets docs method reference: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListBuckets.html